src/main/scala/beam/agentsim/agents/ridehail/RideHailSurgePricingManager.scala

```diff
@@ -10,10 +11,16 @@ import scala.collection.JavaConverters._
 import scala.collection.mutable.ArrayBuffer
 import scala.util.Random
 
-class RideHailSurgePricingManager @Inject() (val beamServices: BeamServices) {
+class RideHailSurgePricingManager @Inject() (
+  val beamConfig: BeamConfig,
+  val beamScenario: BeamScenario,
+  val managerName: String
+) {
```

Comment:
I see. The issue here is that beamServices is initialized after RideHailSurgePricingManager. And since I am initializing a map of multiple RideHailSurgePricingManager objects, I needed to actually initialize each one, which would error without access to beamServices. @nikolayilyin do you have any suggestions for alternative approaches?